gradescopeCronJob/gradescope_to_spreadsheet.py

```diff
@@ -103,13 +105,16 @@ def writeToSheet(sheet_api_instance, assignment_scores, assignment_name = ASSIGN
                     }
                 }
             }
-            response = sheet_api_instance.batchUpdate(spreadsheetId=SPREADSHEET_ID, body=create_sheet_request).execute()
+            request = sheet_api_instance.batchUpdate(spreadsheetId=SPREADSHEET_ID, body=create_sheet_rest_request)
+
+
+            response = make_request(request)
             sheet_id = response['replies'][0]['addSheet']['properties']['sheetId']
         else:
             sheet_id = sub_sheet_titles_to_ids[assignment_name]
-        time.sleep(SLEEP_TIME)
         update_sheet_with_csv(assignment_scores, sheet_api_instance, sheet_id)
-        logger.info(f"Successfully updated spreadsheet with: {assignment_name}")
+        logger.info(f"Successfully updated spreadsheet with: {assignment_name} after {number_of_retries_needed_to_update_sheet} retries")
+        number_of_retries_needed_to_update_sheet = 0
```

Comment:
This function does a batch update for all records, right? If it does a batch update for all records, are we ever reusing this variable `number_of_retries_needed_to_update_sheet `? Cron jobs start with a fresh environment each time, so all variables are lost when the execution is finished, so this line might be doing nothing.

Reply:
It does not do a batch update for all grades; it updates one subsheet (specified by `assignment_name`). We are not re-using `number_of_retries_needed_to_update_sheet` across cron-jobs; it is meant to track the number of retries needed in an attempt to update one sheet.